test/fitnesse/responders/ScopeVariablesResponderTest.java

```diff
@@ -0,0 +1,42 @@
+package fitnesse.responders;
+
+import fitnesse.FitNesseContext;
+import fitnesse.Responder;
+import fitnesse.http.MockRequest;
+import fitnesse.http.SimpleResponse;
+import fitnesse.testutil.FitNesseUtil;
+import fitnesse.wiki.PathParser;
+import fitnesse.wiki.WikiPage;
+import fitnesse.wiki.WikiPageUtil;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static util.RegexTestCase.assertHasRegexp;
+
+public class ScopeVariablesResponderTest {
+  protected WikiPage root;
+  protected MockRequest request;
+  protected Responder responder;
+  protected FitNesseContext context;
+
+  @Before
+  public void setUp() throws Exception {
+    context = FitNesseUtil.makeTestContext();
+    root = context.getRootPage();
+    request = new MockRequest();
+    responder = new ScopeVariablesResponder();
+    WikiPageUtil.addPage(root, PathParser.parse("SimplePageForInclude"), "!define y {Y}");
+    WikiPage page = WikiPageUtil.addPage(root, PathParser.parse("SimplePage"), "!define x {X}\n!include .SimplePageForInclude");
+    WikiPage childPage = WikiPageUtil.addPage(page, PathParser.parse("ChildPage"), "!define z {Z}");
+    WikiPageUtil.addPage(childPage, PathParser.parse("GrandChildPage"), "simple content");
+  }
+
+  @Test
+  public void shouldListVariables() throws Exception {
+    request.setResource("SimplePage.ChildPage.GrandChildPage");
+    SimpleResponse response = (SimpleResponse) responder.makeResponse(context,request);
+    assertHasRegexp("<a href=\"SimplePage\">SimplePage</a>", response.getContent());
+    assertHasRegexp("<a href=\"SimplePage\\.ChildPage\">SimplePage.ChildPage</a>", response.getContent());
+  }
```

Comment:
Test that variable names and values are in the response?

Reply:
Good idea ;) I have extended test in last commit.